File: src/main/java/ru/stnk/RestTestAPI/results/RestResponse.java
package ru.stnk.RestTestAPI.results;

public class RestResponse<T> {

    private T data;
    private int error;
    private String description;

    public RestResponse() {
        this.data = null;
        this.error = 0;
        this.description = "";
    }

    public RestResponse(int error, String description) {
        this.data = null;
        this.error = error;
        this.description = description;
    }

    public RestResponse(T data, int error, String description) {
        this.data = data;
        this.error = error;
        this.description = description;
    }

    public RestResponse(int error) {
        this.data = null;
        this.error = error;
        this.description = "";
    }

    public RestResponse(int error, Throwable ex) {
        this.data = null;
        this.error = error;
        this.description = ex.getMessage();
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public int getError() {
        return error;
    }

    public void setError(int error) {
        this.error = error;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: src/main/kotlin/ru/stnk/resttestapi/results/RestResponse.kt
package ru.stnk.resttestapi.results

import com.fasterxml.jackson.annotation.JsonInclude

// Класс для единого шаблона ответа от сервера
@JsonInclude(JsonInclude.Include.NON_NULL)
class RestResponse {

    var data: Any? = null
    var error: Int = 0
    var description: String? = null

    constructor() {
        this.data = null
        this.error = 0
        this.description = ""
    }

    constructor(error: Int, description: String) {
        this.data = null
        this.error = error
        this.description = description
    }

    constructor(data: Any, error: Int, description: String) {
        this.data = data
        this.error = error
        this.description = description
    }

    constructor(error: Int) {
        this.data = null
        this.error = error
        this.description = ""
    }

    constructor(error: Int, ex: Throwable) {
        this.data = null
        this.error = error
        this.description = ex.message
    }

}
